src/components/modals/FilterModal.tsx

```diff
@@ -0,0 +1,155 @@
+/* eslint-disable no-console */
+// to prevent failed build due to console.log
+// TODO: remove the console log once the backend is ready to accept data
+'use client';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { Button } from '@/components/ui/button';
+import { Form } from '@/components/ui/form';
+import {
+    Dialog,
+    DialogContent,
+    DialogDescription,
+    DialogTitle,
+    DialogTrigger,
+} from '@/components/ui/dialog';
+import { PiSlidersHorizontalBold } from 'react-icons/pi';
+import { useEffect, useState } from 'react';
+import { FilterCheckboxGroup } from '@/components/filters/FilterCheckboxGroup';
+import { FilterState, useFilter } from '@/context/FilterContext';
+
+// Zod Schema for validation
+const FilterSchema = z.object({
+    types: z.array(z.string()).optional(),
+    brands: z.array(z.string()).optional(),
+    priceRanges: z.array(z.string()).optional(),
+    colors: z.array(z.string()).optional(),
+    stock: z.array(z.string()).optional(),
+});
```

Comment:
since the schema for each item is the same, maybe it could be abstracted to some constant or type and reused to limit duplication?